src/cgutils.cpp

```diff
@@ -1286,9 +1338,20 @@ static bool emit_getfield_unknownidx(jl_cgval_t *ret, const jl_cgval_t &strct,
     if (strct.ispointer()) { // boxed or stack
         if (is_datatype_all_pointers(stt)) {
             idx = emit_bounds_check(strct, (jl_value_t*)stt, idx, ConstantInt::get(T_size, nfields), ctx);
-            Value *fld = tbaa_decorate(strct.tbaa, builder.CreateLoad(
-                        builder.CreateGEP(data_pointer(strct, ctx), idx)));
-            if ((unsigned)stt->ninitialized != nfields)
+            bool maybe_null = (unsigned)stt->ninitialized != nfields;
+            size_t minimum_field_size = (size_t)-1;
+            for (size_t i = 0; i < nfields; ++i) {
+                minimum_field_size = std::min(minimum_field_size,
+                    dereferenceable_size(jl_field_type(stt, i)));
+                if (minimum_field_size == 0)
+                    break;
+            }
+            Value *fld = tbaa_decorate(strct.tbaa,
+              maybe_mark_load_dereferenceable(
```

Comment:
no distinction is made between line continuations and control-flow indentation elsewhere - lots of line continuations are indented more than 4 spaces, few if any are indented less